ui/app/components/job-status/deployment-history.js

```diff
@@ -0,0 +1,74 @@
+// @ts-check
+import Component from '@glimmer/component';
+import { alias } from '@ember/object/computed';
+import { tracked } from '@glimmer/tracking';
+import { inject as service } from '@ember/service';
+import { action } from '@ember/object';
+
+export default class JobStatusDeploymentHistoryComponent extends Component {
+  @service notifications;
+
+  /**
+   * @type { Error }
+   */
+  @tracked errorState = null;
+
+  /**
+   * @type { import('../../models/job').default }
+   */
+  @alias('args.deployment.job') job;
+
+  /**
+   * @type { number }
+   */
+  @alias('args.deployment.versionNumber') deploymentVersion;
+
+  /**
+   * Get all allocations for the job
+   * @type { import('../../models/allocation').default[] }
+   */
+  get jobAllocations() {
+    return this.job.get('allocations');
```

Comment:
This does not cause an external data request but just looks against the existing model unless a reload: true flag is passed in.

Having this set up in this way allows us to reference `this.history` and know that it will update with the watched allocations as they roll in.